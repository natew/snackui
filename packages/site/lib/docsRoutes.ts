export const docsRoutes = [
  {
    label: 'Getting Started',
    pages: [
      { title: 'Introduction', slug: 'docs/intro/introduction' },
      { title: 'Installation', slug: 'docs/intro/installation' },
      { title: 'Configuration', slug: 'docs/intro/configuration' },
      { title: 'Themes', slug: 'docs/intro/themes' },
      { title: 'Props', slug: 'docs/intro/props' },
      { title: 'Benchmarks', slug: 'docs/intro/benchmarks' },
      { title: 'Releases', slug: 'docs/intro/releases' },
    ],
  },

  {
    label: 'Tamagui',
    pages: [
      { title: 'Stacks', slug: 'docs/components/stacks' },
      { title: 'Text', slug: 'docs/components/text' },
      { title: 'Button', slug: 'docs/components/button' },
      { title: 'Linear Gradient', slug: 'docs/components/linear-gradient' },
      { title: 'Headings', slug: 'docs/components/headings' },
      { title: 'Visually Hidden', slug: 'docs/components/visually-hidden' },
      { title: 'Forms', slug: 'docs/components/forms' },
      // { title: 'Popover', slug: 'docs/components/popover', pending: true },
      // { title: 'Tooltip', slug: 'docs/components/tooltip', pending: true },
      // { title: 'Grid', slug: 'docs/components/grid' },
    ],
  },

  {
    label: 'Core',
    pages: [
      { title: 'styled', slug: 'docs/core/styled' },
      { title: 'Stack & Text', slug: 'docs/core/stack-and-text' },
      { title: 'Theme', slug: 'docs/core/theme' },
      { title: 'useMedia', slug: 'docs/core/use-media' },
      { title: 'useTheme', slug: 'docs/core/use-theme' },
    ],
  },

  {
    label: 'Guides',
    pages: [
      { title: 'Design Systems', slug: 'docs/guides/design-systems' },
      { title: 'Next.js', slug: 'docs/guides/next-js' },
      { title: 'Expo', slug: 'docs/guides/expo' },
    ],
  },
]

export const allDocsRoutes = docsRoutes.flatMap((x) => x.pages)
