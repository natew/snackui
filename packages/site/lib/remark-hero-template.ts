const fs = require('fs')
const path = require('path')
const utilvisit = require('unist-util-visit')

const ROOT_PATH = process.cwd()
const HERO_PATH = 'components/demos'

export default (options = {}) => {
  return (tree) => {
    utilvisit(tree, 'code', visitor)
  }

  function visitor(node, index, parent) {
    if (!node.meta) return
    const [_, templateName] = node.meta?.split('template=') ?? []
    if (!templateName) return
    const templatePath = path.join(`${ROOT_PATH}/${HERO_PATH}/${templateName}Demo.tsx`)
    let source = fs.readFileSync(path.join(templatePath), 'utf8')
    node.value = source
  }
}
