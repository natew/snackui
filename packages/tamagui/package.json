{
  "name": "tamagui",
  "version": "1.15.30",
  "sideEffects": [
    "*.css",
    "setup.js"
  ],
  "source": "src/index.ts",
  "types": "./types/index.d.ts",
  "main": "dist/cjs",
  "module": "dist/esm",
  "module:jsx": "dist/jsx",
  "files": [
    "src",
    "types",
    "dist",
    "linear-gradient"
  ],
  "scripts": {
    "build": "tamagui-build",
    "watch": "tamagui-build --watch",
    "clean": "tamagui-build clean",
    "clean:build": "tamagui-build clean:build",
    "lint": "../../node_modules/.bin/rome check src",
    "lint:fix": "../../node_modules/.bin/rome check --apply src",
    "check": "yarn check-circular-deps",
    "check-circular-deps": "npx madge --circular ./src/index.ts"
  },
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "types": "./types/index.d.ts",
      "import": "./dist/esm/index.mjs",
      "require": "./dist/cjs/index.js"
    },
    "./linear-gradient": {
      "types": "./types/linear-gradient.d.ts",
      "import": "./dist/esm/linear-gradient.mjs",
      "require": "./dist/cjs/linear-gradient.js"
    }
  },
  "dependencies": {
    "@tamagui/adapt": "1.15.30",
    "@tamagui/alert-dialog": "1.15.30",
    "@tamagui/animate-presence": "1.15.30",
    "@tamagui/avatar": "1.15.30",
    "@tamagui/button": "1.15.30",
    "@tamagui/card": "1.15.30",
    "@tamagui/checkbox": "1.15.30",
    "@tamagui/compose-refs": "1.15.30",
    "@tamagui/core": "1.15.30",
    "@tamagui/create-context": "1.15.30",
    "@tamagui/dialog": "1.15.30",
    "@tamagui/fake-react-native": "1.15.30",
    "@tamagui/focusable": "1.15.30",
    "@tamagui/font-size": "1.15.30",
    "@tamagui/form": "1.15.30",
    "@tamagui/get-button-sized": "1.15.30",
    "@tamagui/get-font-sized": "1.15.30",
    "@tamagui/get-size": "1.15.30",
    "@tamagui/helpers": "1.15.30",
    "@tamagui/helpers-tamagui": "1.15.30",
    "@tamagui/image": "1.15.30",
    "@tamagui/label": "1.15.30",
    "@tamagui/linear-gradient": "1.15.30",
    "@tamagui/list-item": "1.15.30",
    "@tamagui/popover": "1.15.30",
    "@tamagui/popper": "1.15.30",
    "@tamagui/portal": "1.15.30",
    "@tamagui/progress": "1.15.30",
    "@tamagui/radio-group": "1.15.30",
    "@tamagui/react-native-media-driver": "1.15.30",
    "@tamagui/scroll-view": "1.15.30",
    "@tamagui/select": "1.15.30",
    "@tamagui/separator": "1.15.30",
    "@tamagui/shapes": "1.15.30",
    "@tamagui/sheet": "1.15.30",
    "@tamagui/slider": "1.15.30",
    "@tamagui/stacks": "1.15.30",
    "@tamagui/switch": "1.15.30",
    "@tamagui/tabs": "1.15.30",
    "@tamagui/text": "1.15.30",
    "@tamagui/theme": "1.15.30",
    "@tamagui/toggle-group": "1.15.30",
    "@tamagui/tooltip": "1.15.30",
    "@tamagui/use-controllable-state": "1.15.30",
    "@tamagui/use-debounce": "1.15.30",
    "@tamagui/use-event": "1.15.30",
    "@tamagui/use-force-update": "1.15.30",
    "@tamagui/use-window-dimensions": "1.15.30",
    "@tamagui/visually-hidden": "1.15.30",
    "reforest": "^0.12.1"
  },
  "peerDependencies": {
    "react": "*",
    "react-native-web": "*"
  },
  "devDependencies": {
    "@tamagui/build": "1.15.30",
    "@types/node": "^16.11.9",
    "@types/react": "^18.0.15",
    "react": "^18.2.0",
    "react-native-web": "^0.18.10"
  },
  "gitHead": "a49cc7ea6b93ba384e77a4880ae48ac4a5635c14"
}
