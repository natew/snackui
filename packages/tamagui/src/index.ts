import './polyfills'

export * from '@tamagui/animate-presence'
export * from '@tamagui/alert-dialog'
export * from '@tamagui/avatar'
export * from '@tamagui/button'
export * from '@tamagui/card'
export * from '@tamagui/compose-refs'
export * from '@tamagui/create-context'
export * from '@tamagui/dialog'
export * from '@tamagui/font-size'
export * from '@tamagui/helpers-tamagui'
export * from '@tamagui/image'
export * from '@tamagui/label'
export * from '@tamagui/list-item'
export * from '@tamagui/popover'
export * from '@tamagui/popper'
export * from '@tamagui/portal'
export * from '@tamagui/progress'
export * from '@tamagui/scroll-view'
export * from '@tamagui/select'
export * from '@tamagui/separator'
export * from '@tamagui/shapes'
export * from '@tamagui/sheet'
export * from '@tamagui/slider'
export * from '@tamagui/stacks'
export * from '@tamagui/switch'
export * from '@tamagui/text'
export * from '@tamagui/tooltip'
export * from '@tamagui/use-controllable-state'
export * from '@tamagui/use-debounce'
export * from '@tamagui/use-window-dimensions'
export * from '@tamagui/use-force-update'
export * from '@tamagui/visually-hidden'

export * from './views/TamaguiProvider'
export * from './viewTypes'

export * from './views/Anchor'
export * from './views/EnsureFlexed'
export * from './views/Fieldset'
export * from './views/Form'
export * from './views/Grid'
export * from './views/Input'
export * from './views/Group'
export * from './views/Layouts'
export * from './views/LinearGradient'
// export * from './views/Skeleton'
export * from './views/Spinner'
export * from './views/TextArea'

// since we overlap with StackProps and potentially others
// lets be explicit on what gets exported
export type {
  AnimationKeys,
  ColorTokens,
  CreateTamaguiConfig,
  CreateTamaguiProps,
  FontColorTokens,
  FontLetterSpacingTokens,
  FontLineHeightTokens,
  FontSizeTokens,
  FontStyleTokens,
  FontTokens,
  FontTransformTokens,
  FontWeightTokens,
  GenericFont,
  GenericTamaguiConfig,
  GetAnimationKeys,
  GetProps,
  GetVariantProps,
  GenericTextVariants,
  GenericStackVariants,
  Media,
  MediaPropKeys,
  MediaQueries,
  MediaQueryState,
  RNWTextProps,
  RNWViewProps,
  Shorthands,
  Slot,
  Slottable,
  SizeTokens,
  SpaceTokens,
  SpacerProps,
  StackPropsBase,
  TamaguiElement,
  StackPropsBaseShared,
  StaticConfig,
  StaticConfigParsed,
  TamaguiBaseTheme,
  TamaguiComponent,
  TamaguiComponentPropsBase,
  TamaguiConfig,
  FontLanguages,
  TamaguiCustomConfig,
  TamaguiInternalConfig,
  TamaguiProviderProps,
  TextProps,
  TextPropsBase,
  TextPropsBaseShared,
  ThemeKeys,
  ThemeName,
  ThemeParsed,
  ThemeProps,
  Themes,
  ThemeValueFallback,
  Tokens,
  TransformStyleProps,
  VariantSpreadExtras,
  VariantSpreadFunction,
  ZIndexTokens,
} from '@tamagui/core'

export {
  createTamagui,
  // components
  Spacer,
  Stack,
  Text,
  Theme,
  ThemeInverse,
  ThemeReset,
  FontLanguage,
  Unspaced,
  // context,
  TextAncestorContext,
  // constants
  isChrome,
  isClient,
  isServer,
  isTouchable,
  isWeb,
  isWebTouchable,
  // helpers
  calc,
  addMediaQueryListener,
  removeMediaQueryListener,
  mediaQueryConfig,
  createComponent,
  createFont,
  createShorthands,
  createTheme,
  addTheme,
  updateTheme,
  createTokens,
  createVariable,
  getConfig,
  getAnimationDriver,
  getHasConfigured,
  getMedia,
  getStylesAtomic,
  getThemes,
  getTokens,
  getVariable,
  variableToString,
  getVariableValue,
  getVariableName,
  insertFont,
  isObj,
  isTamaguiElement,
  isTamaguiComponent,
  isVariable,
  matchMedia,
  mediaObjectToString,
  mediaState,
  spacedChildren,
  styled,
  themeable,
  withStaticProperties,
  // hooks
  useConstant,
  useGet,
  useEvent,
  useDefaultThemeName,
  useIsMounted,
  useIsomorphicLayoutEffect,
  useIsTouchDevice,
  useMedia,
  usePressable,
  useTheme,
  useThemeName,
  useSafeRef,
  useUnmountEffect,
} from '@tamagui/core'
