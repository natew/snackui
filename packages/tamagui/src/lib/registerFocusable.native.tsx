// used for focusing on native

type Focusable = { focus: Function }

const InputsMap = new Map<string, Focusable>()

export const registerFocusable = (id: string, input: Focusable) => {
  if (InputsMap.has(id)) {
    console.warn(`Warning, duplicate ID for input: ${id}`)
  }
  InputsMap.set(id, input)
  return () => {
    InputsMap.delete(id)
  }
}

export const unregisterFocusable = (id: string) => {
  InputsMap.delete(id)
}

export const focusFocusable = (id: string) => {
  const input = InputsMap.get(id)
  input?.focus()
}
