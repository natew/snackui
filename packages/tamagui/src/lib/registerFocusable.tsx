export const registerFocusable = (id: string, input: { focus: Function }) => {
  // noop focus is handed natively
}

export const unregisterFocusable = (id: string) => {
  // noop focus is handed natively
}

export const focusFocusable = (id: string) => {
  // noop focus is handed natively
}
