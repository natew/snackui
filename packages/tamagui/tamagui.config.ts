import { config } from '@tamagui/config'
import { createTamagui } from '@tamagui/core'

export default createTamagui(config)
