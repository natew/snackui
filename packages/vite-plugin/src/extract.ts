// fork from https://github.com/seek-oss/vanilla-extract

import path from 'path'

import { TamaguiOptions } from '@tamagui/static'
import * as Static from '@tamagui/static'
import outdent from 'outdent'
import type { Plugin, ResolvedConfig, ViteDevServer } from 'vite'
import { normalizePath } from 'vite'

const { createExtractor, extractToClassNames, getPragmaOptions } =
  Static['default'] || Static

const styleUpdateEvent = (fileId: string) => `tamagui-style-update:${fileId}`
const GLOBAL_CSS_VIRTUAL_PATH = '__tamagui_global_css__.css'

export function tamaguiExtractPlugin(options: TamaguiOptions): Plugin {
  const disableStatic =
    options.disable || (options.disableDebugAttr && options.disableExtraction)

  if (disableStatic) {
    return {
      name: 'tamagui-extract',
    }
  }

  let extractor: ReturnType<typeof createExtractor> | null = null
  const cssMap = new Map<string, string>()

  let config: ResolvedConfig
  let server: ViteDevServer
  let shouldReturnCSS = true //config.command === 'serve'
  let virtualExt: string

  const getAbsoluteVirtualFileId = (filePath: string) => {
    if (filePath.startsWith(config.root)) {
      return filePath
    }
    return normalizePath(path.join(config.root, filePath))
  }

  return {
    name: 'tamagui-extract',
    enforce: 'pre',

    configureServer(_server) {
      server = _server
    },

    buildEnd() {
      extractor!.cleanupBeforeExit()
    },

    writeBundle(this, options, bundle) {
      setTimeout(() => {
        // rome-ignore lint/nursery/noConsoleLog: ok
        console.warn('some sort of dangling process or osmethign, exit for now...')
        process.exit(0)
      }, 100)
    },

    config(_userConfig, env) {
      const include = env.command === 'serve' ? ['@tamagui/core/inject-styles'] : []
      return {
        optimizeDeps: { include },
      }
    },

    async configResolved(resolvedConfig) {
      config = resolvedConfig
      extractor = createExtractor({
        logger: resolvedConfig.logger,
      })
      shouldReturnCSS = true
      // TODO postcss work with postcss.config.js
      // packageName = getPackageInfo(config.root).name;
      // if (config.command === 'serve') {
      //   postCssConfig = await resolvePostcssConfig(config);
      // }
      virtualExt = `.tamagui.${shouldReturnCSS ? 'css' : 'js'}`
    },

    async resolveId(source) {
      if (source === 'tamagui.css') {
        await extractor!.loadTamagui(options)
        return GLOBAL_CSS_VIRTUAL_PATH
      }

      const [validId, query] = source.split('?')

      if (!validId.endsWith(virtualExt)) {
        return
      }

      // Absolute paths seem to occur often in monorepos, where files are
      // imported from outside the config root.
      const absoluteId = source.startsWith(config.root)
        ? source
        : getAbsoluteVirtualFileId(validId)

      // There should always be an entry in the `cssMap` here.
      // The only valid scenario for a missing one is if someone had written
      // a file in their app using the .tamagui.js/.tamagui.css extension
      if (cssMap.has(absoluteId)) {
        // Keep the original query string for HMR.
        return absoluteId + (query ? `?${query}` : '')
      }
    },

    /**
     * TODO
     *
     *   mainFields module:jsx breaks, so lets just have a mapping here
     *   where we load() and map it to the jsx path before transform
     *
     */

    load(id, options) {
      const [validId] = id.split('?')

      if (validId === GLOBAL_CSS_VIRTUAL_PATH) {
        return extractor!.getTamagui()!.getCSS()
      }

      if (!cssMap.has(validId)) {
        return
      }

      const css = cssMap.get(validId)

      if (typeof css !== 'string') {
        return
      }

      if (shouldReturnCSS || !server || server.config.isProduction) {
        return css
      }

      return outdent`
        import { injectStyles } from '@tamagui/core/inject-styles';

        const inject = (css) => injectStyles({
          filePath: "${validId}",
          css
        });

        inject(${JSON.stringify(css)});

        if (import.meta.hot) {
          import.meta.hot.on('${styleUpdateEvent(validId)}', (css) => {
            inject(css);
          });
        }
      `
    },

    async transform(code, id, ssrParam) {
      const [validId] = id.split('?')

      if (!validId.endsWith('.tsx')) {
        return
      }

      // let ssr: boolean | undefined
      // if (typeof ssrParam === 'boolean') {
      //   ssr = ssrParam
      // } else {
      //   ssr = ssrParam?.ssr
      // }

      const firstCommentIndex = code.indexOf('// ')
      const { shouldDisable, shouldPrintDebug } = getPragmaOptions({
        source: firstCommentIndex >= 0 ? code.slice(firstCommentIndex) : '',
        path: validId,
      })

      if (shouldDisable) {
        return
      }

      const extracted = await extractToClassNames({
        extractor: extractor!,
        source: code,
        sourcePath: validId,
        options,
        shouldPrintDebug,
      })

      if (!extracted) {
        return
      }

      const rootRelativeId = `${validId}${virtualExt}`
      const absoluteId = getAbsoluteVirtualFileId(rootRelativeId)

      let source = extracted.js

      if (extracted.styles) {
        if (
          server &&
          cssMap.has(absoluteId) &&
          cssMap.get(absoluteId) !== extracted.styles
        ) {
          const { moduleGraph } = server
          const [module] = Array.from(moduleGraph.getModulesByFile(absoluteId) || [])

          if (module) {
            moduleGraph.invalidateModule(module)

            // Vite uses this timestamp to add `?t=` query string automatically for HMR.
            module.lastHMRTimestamp =
              (module as any).lastInvalidationTimestamp || Date.now()
          }

          server.ws.send({
            type: 'custom',
            event: styleUpdateEvent(absoluteId),
            data: extracted.styles,
          })
        }

        source = `${source}\nimport "${rootRelativeId}";`
        cssMap.set(absoluteId, extracted.styles)
      }

      return {
        code: source.toString(),
        map: extracted.map,
      }

      // if (ssr && !process.env.VITE_RSC_BUILD) {
      //   return addFileScope({
      //     source: code,
      //     filePath: normalizePath(validId),
      //     rootPath: config.root,
      //     packageName,
      //   })
      // }

      // const { source, watchFiles } = await compile({
      //   filePath: validId,
      //   cwd: config.root,
      //   esbuildOptions,
      // })

      // for (const file of watchFiles) {
      //   // In start mode, we need to prevent the file from rewatching itself.
      //   // If it's a `build --watch`, it needs to watch everything.
      //   if (config.command === 'build' || file !== validId) {
      //     this.addWatchFile(file)
      //   }
      // }
    },
  }
}
