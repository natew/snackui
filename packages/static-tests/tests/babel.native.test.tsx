import * as React from 'react'
import { expect, test } from 'vitest'

import { extractForNative, extractForWeb } from './lib/extract'

process.env.TAMAGUI_TARGET = 'native'
process.env.IS_STATIC = ''

window['React'] = React

test('basic extraction', async () => {
  const output = await extractForNative(`
    import { YStack } from 'tamagui'
    export function Test() {
      return (
        <YStack backgroundColor="red" />
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test('theme value extraction should work when no theme variables used', async () => {
  // here we override default "$color" so it should flatten totally
  const output = await extractForNative(`
    import { Paragraph } from 'tamagui'
    export function Test() {
      return (
        <Paragraph color="red">hello world</Paragraph>
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test('theme value extraction should NOT work when theme variables used', async () => {
  // here we DO NOT override default "$color" so it WONT flatten
  const output = await extractForNative(`
    import { Paragraph } from 'tamagui'
    export function Test() {
      return (
        <Paragraph>hello world</Paragraph>
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test('basic conditional extraction', async () => {
  const output = await extractForNative(`
    import { YStack } from 'tamagui'
    export function Test() {
      return (
        <>
          <YStack backgroundColor={x ? 'red' : 'blue'} />
          <YStack {...x && { backgroundColor: 'red' }} />
        </>
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test('flat transform props', async () => {
  const output = await extractForNative(`
    import { YStack } from 'tamagui'
    export function Test(isLoading) {
      return (
        <YStack
          scale={isLoading ? 1 : 2}
          x={10}
          y={20}
          rotate="10deg"
        />
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test('handles style order merge properly', async () => {
  const output = await extractForNative(`
    import { YStack } from 'tamagui'
    export function Test(props) {
      return (
        <YStack
          scale={props.isLoading ? 1 : 2}
          x={10}
          {...props}
          rotate="10deg"
        />
      )
    }
  `)
  const code = output?.code ?? ''
  expect(code).toMatchSnapshot()
})

test(`normalize ternaries flips the conditional properly`, async () => {
  const inputCode = `
  import { Stack } from 'tamagui'
  export function Test(props) {
    return (
      <Stack marginBottom={props !== 123 ? 12 : 0} />
    )
  }
`
  const output = await extractForNative(inputCode)
  const outCode = output?.code ?? ''
  expect(outCode).toContain(`props === 123 ? _sheet["1"] : _sheet["2"]`)
  expect(outCode).toContain(`  "1": {
    "marginBottom": 0
  },
  "2": {
    "marginBottom": 12
  }`)
})
