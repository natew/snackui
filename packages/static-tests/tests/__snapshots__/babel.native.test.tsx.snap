// Vitest Snapshot v1

exports[`basic conditional extraction 1`] = `
"import { StyleSheet as ReactNativeStyleSheet } from 'react-native';
const _sheet = ReactNativeStyleSheet.create({
  \\"0\\": {},
  \\"1\\": {
    \\"alignItems\\": \\"stretch\\",
    \\"flexShrink\\": 0,
    \\"display\\": \\"flex\\",
    \\"flexBasis\\": \\"auto\\",
    \\"position\\": \\"relative\\",
    \\"minHeight\\": 0,
    \\"minWidth\\": 0,
    \\"flexDirection\\": \\"column\\"
  },
  \\"2\\": {
    \\"backgroundColor\\": \\"rgba(255,0,0,1.00)\\"
  },
  \\"3\\": {
    \\"backgroundColor\\": \\"rgba(0,0,255,1.00)\\"
  },
  \\"4\\": {},
  \\"5\\": {
    \\"alignItems\\": \\"stretch\\",
    \\"flexShrink\\": 0,
    \\"display\\": \\"flex\\",
    \\"flexBasis\\": \\"auto\\",
    \\"position\\": \\"relative\\",
    \\"minHeight\\": 0,
    \\"minWidth\\": 0,
    \\"flexDirection\\": \\"column\\"
  },
  \\"6\\": {
    \\"backgroundColor\\": \\"rgba(255,0,0,1.00)\\"
  },
  \\"7\\": {}
});
import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { YStack } from 'tamagui';
export function Test() {
  return <>
          <__ReactNativeView style={[_sheet[\\"0\\"], _sheet[\\"1\\"], x ? _sheet[\\"2\\"] : _sheet[\\"3\\"]]} />
          <__ReactNativeView style={[_sheet[\\"4\\"], _sheet[\\"5\\"], x ? _sheet[\\"6\\"] : _sheet[\\"7\\"]]} />
        </>;
}"
`;

exports[`basic extraction 1`] = `
"import { StyleSheet as ReactNativeStyleSheet } from 'react-native';
const _sheet = ReactNativeStyleSheet.create({
  \\"0\\": {},
  \\"1\\": {
    \\"alignItems\\": \\"stretch\\",
    \\"flexShrink\\": 0,
    \\"display\\": \\"flex\\",
    \\"flexBasis\\": \\"auto\\",
    \\"position\\": \\"relative\\",
    \\"minHeight\\": 0,
    \\"minWidth\\": 0,
    \\"flexDirection\\": \\"column\\",
    \\"backgroundColor\\": \\"rgba(255,0,0,1.00)\\"
  }
});
import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { YStack } from 'tamagui';
export function Test() {
  return <__ReactNativeView style={[_sheet[\\"0\\"], _sheet[\\"1\\"]]} />;
}"
`;

exports[`flat transform props 1`] = `
"import { StyleSheet as ReactNativeStyleSheet } from 'react-native';
const _sheet = ReactNativeStyleSheet.create({
  \\"0\\": {},
  \\"1\\": {
    \\"alignItems\\": \\"stretch\\",
    \\"flexShrink\\": 0,
    \\"display\\": \\"flex\\",
    \\"flexBasis\\": \\"auto\\",
    \\"position\\": \\"relative\\",
    \\"minHeight\\": 0,
    \\"minWidth\\": 0,
    \\"flexDirection\\": \\"column\\"
  },
  \\"2\\": {
    \\"transform\\": [{
      \\"scale\\": 1
    }]
  },
  \\"3\\": {
    \\"transform\\": [{
      \\"scale\\": 2
    }]
  },
  \\"4\\": {
    \\"transform\\": [{
      \\"translateY\\": 20
    }, {
      \\"translateX\\": 10
    }, {
      \\"rotate\\": \\"10deg\\"
    }]
  }
});
import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { YStack } from 'tamagui';
export function Test(isLoading) {
  return <__ReactNativeView style={[_sheet[\\"0\\"], _sheet[\\"1\\"], isLoading ? _sheet[\\"2\\"] : _sheet[\\"3\\"], _sheet[\\"4\\"]]} />;
}"
`;

exports[`handles style order merge properly 1`] = `
"import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { YStack } from 'tamagui';
export function Test(props) {
  return <YStack scale={props.isLoading ? 1 : 2} x={10} {...props} rotate=\\"10deg\\" />;
}"
`;

exports[`normalize ternaries flips the conditional properly 1`] = `
"import { StyleSheet as ReactNativeStyleSheet } from 'react-native';
const _sheet = ReactNativeStyleSheet.create({
  \\"0\\": {},
  \\"1\\": {
    \\"alignItems\\": \\"stretch\\",
    \\"flexShrink\\": 0,
    \\"display\\": \\"flex\\",
    \\"flexBasis\\": \\"auto\\",
    \\"position\\": \\"relative\\",
    \\"minHeight\\": 0,
    \\"minWidth\\": 0,
    \\"flexDirection\\": \\"column\\"
  },
  \\"2\\": {
    \\"marginBottom\\": 0
  },
  \\"3\\": {
    \\"marginBottom\\": 12
  }
});
import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { Stack } from 'tamagui';
export function Test(props) {
  return <__ReactNativeView style={[_sheet[\\"0\\"], _sheet[\\"1\\"], props === 123 ? _sheet[\\"2\\"] : _sheet[\\"3\\"]]} />;
}"
`;

exports[`theme value extraction should NOT work when theme variables used 1`] = `
"import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { Paragraph } from 'tamagui';
export function Test() {
  return <Paragraph>hello world</Paragraph>;
}"
`;

exports[`theme value extraction should work when no theme variables used 1`] = `
"import { View as __ReactNativeView, Text as __ReactNativeText } from 'react-native';
import { Paragraph } from 'tamagui';
export function Test() {
  return <Paragraph color=\\"red\\">hello world</Paragraph>;
}"
`;
